code/game/machinery/computer/law.dm

```diff
@@ -36,9 +46,45 @@
 		if(!atoms_share_level(T, src))
 			to_chat(user, "<span class='danger'>Unable to establish a connection</span>: You're too far away from the target silicon!")
 			return
-		var/obj/item/aiModule/M = O
-		M.install(src)
-		return
+		if (!emagged) //non-emag law change
+			var/obj/item/aiModule/M = O
+			M.install(src)
+			return
+		else
+			if(world.time < cooldown)
+				to_chat(user, "The program seems to have frozen. It will need some time to process")
+				return
+			do_sparks(5, 1, src)
+			var/foundlaws = 0
+			var/checked = FALSE
+			for(var/datum/ai_law/law in current.laws.all_laws())
+				if (law in current.laws.inherent_laws)
+					foundlaws = foundlaws + 1
+				if(current.laws.ion_laws.len == 0 && checked == FALSE)
+					checked = TRUE
+					if (rand(1,10) >= 9)  // 20% chance to generate an ion law if none exists
+						current.add_ion_law(generate_ion_law())
+						cooldown = world.time + 600
```

Comment:
Turn the "600" into a cooldown define.

Reply:
fixed and used var/cooldown_delay = 60 SECONDS